share/lib/hoc/import3d/import3d_gui.hoc

```diff
@@ -1237,6 +1237,9 @@ proc contour2centroid() {local i, j, imax, imin, ok  localobj mean, pts, d, max,
 	// minor is normal and in xy plane
 	minor = m.getcol(3-tobj.min_ind-tobj.max_ind)
 	minor.x[2] = 0
+	if (minor.mag == 0.) {
```

Comment:
Likely, `minor.mag == 1e-203` is just as much of a sign that something is off. We should replace this with:
```
if (abs(minor.mag) < EPS) {
```
However, the question is what's a sensible values for `EPS`. It depends on how large `minor.mag` "usually" is, i.e. some notion of characteristic size.

Reply:
Thank @nrnhines @1uc  for the comments. 
So shall I throw when ```if (minor.mag/major.mag < 1e-6)```?

Reply:
> So shall I throw when `if (minor.mag/major.mag < 1e-6)`?

That is acceptable to me.